chia/full_node/mempool.py

```diff
@@ -471,12 +473,52 @@ def at_full_capacity(self, cost: int) -> bool:
 
         return self._total_cost + cost > self.mempool_info.max_size_in_cost
 
+    async def create_block_generator(
+        self,
+        get_unspent_lineage_info_for_puzzle_hash: Callable[[bytes32], Awaitable[Optional[UnspentLineageInfo]]],
+        constants: ConsensusConstants,
+        height: uint32,
+        item_inclusion_filter: Optional[Callable[[bytes32], bool]] = None,
+    ) -> tuple[BlockGenerator, G2Element, list[Coin]]:
+        """
+        height is needed in case we fast-forward a transaction and we need to
+        re-run its puzzle.
+        """
+
+        mempool_bundle = await self.create_bundle_from_mempool_items(
+            get_unspent_lineage_info_for_puzzle_hash, constants, height, item_inclusion_filter
+        )
+        if mempool_bundle is None:
+            return (BlockGenerator(), G2Element(), [])
+
+        spend_bundle, additions = mempool_bundle
+        removals = spend_bundle.removals()
+        log.info(f"Add rem: {len(additions)} {len(removals)}")
+
+        # since the hard fork has activated, block generators are
+        # allowed to be serialized with CLVM back-references. We can do that
+        # unconditionally.
+        start_time = monotonic()
+        spends = [(cs.coin, bytes(cs.puzzle_reveal), bytes(cs.solution)) for cs in spend_bundle.coin_spends]
+        block_program = solution_generator_backrefs(spends)
+
+        duration = monotonic() - start_time
+        log.log(
+            logging.INFO if duration < 1 else logging.WARNING,
+            f"serializing block generator took {duration:0.2f} seconds",
+        )
```

Comment:
If we want to continue measuring the block generator serialization duration accurately, we need to keep using `simple_solution_generator_backrefs` here.

Reply:
why?

Reply:
The current behavior tracks the duration of calling `simple_solution_generator_backrefs` which contains the deserialization step and the construction of BlockGenerator, but with this, we're tracking the duration of one less step which got moved to the return step: `BlockGenerator(SerializedProgram.from_bytes(block_program), [])`.